backend/tests/test_logging_axiom.py

```
@@ -0,0 +1,564 @@
+
+class TestHttpErrorHandling:
+    """4xx / 5xx responses must produce a status-only stderr warning and no exception."""
+
+    def test_4xx_response_warns_but_does_not_raise(self, capsys):
+        handler = AxiomHandler(token="t", dataset="d")
+        try:
+            fake_response = MagicMock()
+            fake_response.status_code = 403
+            # If anyone tries to read the body, fail loudly so the test catches a regression.
+            fake_response.text = property(  # type: ignore[assignment]
+                lambda self: pytest.fail("response body must not be read")
```

Ruff (0.15.13)

[warning] Unused lambda argument: `self`

(ARG005)